broker.go

```diff
@@ -358,6 +372,33 @@ func (b *Broker) fetchMetadata(topics ...string) (*proto.MetadataResp, error) {
 		return resp, nil
 	}
 
+	var resp *proto.MetadataResp
+
+	ok := b.dialRun(func(conn *connection, addr string) bool {
```

Comment:
I think it is.  On the previous step, we were trying to get info from cached b.metadata.nodes.   There is a possibility that this list doesn't contain any valid node ( or any node at all) .  So on this step, we try to connect to any random node we know.   It's like a backup way to update metadata and I wanted to keep it simple.